src/types/routeFlat.ts

```diff
@@ -1,5 +1,9 @@
+import { Param } from '@/types/params'
+import { Path } from '@/utilities'
+
 export type RouteFlat = {
   name: string,
-  path: string,
+  path: string | Path,
   regex: RegExp,
+  params: Record<string, Param[]>,
```

Comment:
should use `ExtractParamsFromPath<T>`, but with new type for `P` since all params will be populated now (defaults to String)